client/allocator/src/freeing_bump.rs

```diff
@@ -601,21 +642,65 @@ impl<'a> Drop for PoisonBomb<'a> {
 mod tests {
 	use super::*;
 
-	const PAGE_SIZE: u32 = 65536;
-
 	/// Makes a pointer out of the given address.
 	fn to_pointer(address: u32) -> Pointer<u8> {
 		Pointer::new(address)
 	}
 
+	struct MemoryInstance {
+		data: Vec<u8>,
+		max_wasm_pages: u32,
+	}
+
+	impl MemoryInstance {
+		fn new() -> Self {
+			Self { data: vec![0; PAGE_SIZE as usize], max_wasm_pages: MAX_WASM_PAGES }
+		}
```

Comment:
I'd probably remove this and just use `with_size` everywhere in tests to be explicit. (When reading the code I first assumed that `new` will return a zero-sized memory since literally every Rust data structure works like that [except VecDeque])